### ISSUE TITLE
SC2093: Suppress warning if exec is followed by return

### ISSUE THREAD

**Issue by ericvw** [koalaman/shellcheck#2249]
#### For bugs
- Rule Id (if any, e.g., SC1000): SC2093
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2
- [X] The rule's wiki page does not already cover this (https://github.com/koalaman/shellcheck/wiki/SC2093)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bin
# This file is being sourced by another script.

exec thisdoesnotexist
echo "Error invoking doesnotexist"
return
```

#### Here's what shellcheck currently says:

```
$ shellcheck t.sh
In t.sh line 4:
exec thisdoesnotexist
^-------------------^ SC2093: Remove "exec " if script should continue after this command.

For more information:
  https://www.shellcheck.net/wiki/SC2093 -- Remove "exec " if script should c...
  ```

#### Here's what I wanted or expected to see:

```
$ shellcheck t.sh
$
```

Suppressing this warning is useful in contexts when `exec` is followed by a `return` in a sourced file where you want to allow graceful completion of the parent script (e.g., shell initialization; `exit` would prevent the login shell from coming up).

**Comment by koalaman** [comment]
`return` should indeed be allowed for the same reason as the current `exit`, and now it is. Thanks!

The same change also adds `printf` for the same reason as `echo`, and now ignores assignments to allow `error=$?` and such.
